pkg/unlock.go

```diff
@@ -71,62 +60,70 @@ func NewCmdUnlockRepository(clientGetter genericclioptions.RESTClientGetter) *co
 			if err != nil {
 				return err
 			}
-			// unlock local backend
-			if repository.Spec.Backend.Local != nil {
-				if err = unlockLocalRepo(kc, repository); err != nil {
-					return fmt.Errorf("can't unlock repository for local backend, reason: %s", err)
-				}
-				return nil
-			}
+
 			// get source repository secret
 			secret, err := kc.CoreV1().Secrets(namespace).Get(repository.Spec.Backend.StorageSecretName, metav1.GetOptions{})
 			if err != nil {
 				return err
 			}
 
+			tempDir, err := ioutil.TempDir("", "stash-cli")
+			if err != nil {
+				return err
+			}
+			defer os.RemoveAll(tempDir)
+
+			// dump secret
```

Comment:
// dump secret into a temporary directory.
// we will pass the secret files into restic docker container.